### PR TITLE
Optimized ReadOnlyBytes

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
Optimized some more base data structures, mainly ReadOnlyBytes.Slice(int)
This change resulted in ~10% improvement in the E2EPipelineTests benchmarks.

cc: @pakrym, @davidfowl, @shiftylogic, @ahsonkhan

### CODE REVIEW

File: src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs [hunk 1]
```diff
@@ -104,21 +104,31 @@ public ReadOnlyBytes Slice(Range range)
...
             {
-                return new ReadOnlyBytes(First.Slice(index, length));
+                return new ReadOnlyBytes(first.Slice(index, length));
             }
-            if (First.Length > index)
+            if (first.Length > index)
             {
-                return new ReadOnlyBytes(_first.Slice(index), _rest, length);
+                return new ReadOnlyBytes(first.Slice(index), _rest, length);
             }
```
Comment (ahsonkhan) [inline src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs]: This used to be a member variable _first, and now it is a local variable first ( = First). Was it incorrect before?

File: src/System.Slices/System/Buffers/OwnedMemory.cs [hunk 2]
```diff
@@ -161,29 +156,20 @@ internal unsafe bool TryGetPointerInternal(long id, out void* pointer)
...
-        }
-
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal Span<T> GetSpanInternal(long id, int index, int length)
         {
             VerifyId(id);
             var array = Array;
-            if (array != null)
+            if (array != null) {
                 return new Span<T>(array, Offset + index, length);
```
Comment (ahsonkhan) [inline src/System.Slices/System/Buffers/OwnedMemory.cs]: Should brace brackets be on a new line?

File: src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs (hunk 1 above)
Comment (KrzysztofCwalina) [inline src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs]: First == _first == first

File: src/System.Slices/System/Buffers/OwnedMemory.cs (hunk 2 above)
Comment (KrzysztofCwalina) [inline src/System.Slices/System/Buffers/OwnedMemory.cs]: I thought we put brackets on newline for methods and types and on the same line for control blocks. If not, i will change it.

**Comment by KrzysztofCwalina** [comment]
@dotnet-bot test this please

File: src/System.Slices/System/Buffers/OwnedMemory.cs (hunk 2 above)
Comment (benaadams) [inline src/System.Slices/System/Buffers/OwnedMemory.cs]: You :trollface: surely 😉

Comment (ahsonkhan) [inline src/System.Slices/System/Buffers/OwnedMemory.cs]: I am confused. Is that a yes? Should we have loop/branch blocks have brace brackets on new lines? All the code I have touched I have been using new line, which is why I noticed this.

File: src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs (hunk 1 above)
Comment (ahsonkhan) [inline src/System.Buffers.Experimental/System/Buffers/ReadOnlyBytes.cs]: Got it. Thanks.

**Pull request merged**